zipline/utils/calendars/calendar_utils.py

```diff
@@ -127,21 +141,91 @@ def register_calendar_type(self, name, calendar_type, force=False):
             The type of the calendar to register.
         force : bool, optional
             If True, old calendars will be overwritten on a name collision.
-            If False, name collisions will raise an exception. Default: False.
+            If False, name collisions will raise an exception.
+            Default is False.
 
         Raises
         ------
         CalendarNameCollision
             If a calendar is already registered with the given calendar's name.
         """
         if force:
-            self._calendar_factories.pop(name, None)
+            self.deregister_calendar(name)
 
-        if name in self._calendars or name in self._calendar_factories:
+        if self.has_calendar(name):
             raise CalendarNameCollision(calendar_name=name)
 
         self._calendar_factories[name] = calendar_type
 
+    def register_calendar_alias(self, alias, real_name, force=False):
+        """
+        Register an alias for a calendar.
+
+        This is useful when multiple exchanges should share a calendar, or when
+        there are multiple ways to refer to the same exchange.
+
+        After calling ``register_alias('alias', 'real_name')``, subsequent
+        calls to ``get_calendar('alias')`` will return the same result as
+        ``get_calendar('real_name')``.
+
+        Parameters
+        ----------
+        alias : str
+            The name to be used to refer to a calendar.
+        real_name : str
+            The canonical name of the registered calendar.
+        force : bool, optional
+            If True, old calendars will be overwritten on a name collision.
+            If False, name collisions will raise an exception.
+            Default is False.
+        """
+        if force:
+            self.deregister_calendar(alias)
+
+        if self.has_calendar(alias):
+            raise CalendarNameCollision(calendar_name=alias)
+
+        self._aliases[alias] = real_name
+
+        # Ensure that the new alias doesn't create a cycle, and back it out if
+        # we did.
+        try:
+            self.resolve_alias(alias)
+        except CyclicCalendarAlias:
+            del self._aliases[alias]
+            raise
+
+    def resolve_alias(self, name):
+        """
+        Resolve a calendar alias for retrieval.
+
+        Parameters
+        ----------
+        name : str
+            The name of the requested calendar.
+
+        Returns
+        -------
+        canonical_name : str
+            The real name of the calendar to create/return.
+        """
+        # Use an OrderedDict as an ordered set so that we can return the order
+        # of aliases in the event of a cycle.
+        seen = OrderedDict({name: None})
```

Comment:
Are you just using this as an ordered set? Could we just use a list here, I can't imagine this growing very large